tests/travis-common-script.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+set -e
+
+function main() {
+    . tests/build-log-cflags-propagation.sh;
+    if [ "$CC" = "gcc" ]; then
+      export DISTCHECK_CONFIGURE_FLAGS="$CONFIGURE_FLAGS";
+      exec_prop_check "make distcheck -j 3 V=1 --keep-going";
+      find . -name test-suite.log | xargs cat;
+    else
+      make --keep-going -j $(nproc);
+      S=$?;
+      if [ "$S" = "0" ]; then
+        make install
+        . scripts/get-libjvm-path.sh || return $?;
+        export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$JNI_LIBDIR";
+        make func-test V=1;
+        make pytest-self-check;
+        make pytest-check;
+      elif [ "$S" = "42" ]; then
+        return $S;
+      else
+        make V=1 install;
+        return $S;
+      fi;
+    fi
+}
```

Comment:
Please do not hide this complexity, it does not help IMHO.

We should clean up this mess.
In my opinion, our build steps should be listed directly in travis.yml, in a simple and sequential way, such as:

```

gcc-job:
- . tests/build-log-cflags-propagation.sh;
  exec_prop_check "make distcheck -j 3 V=1 --keep-going";
- print_test_logs

clang-job:
- make --keep-going -j $(nproc);
- make install
- setup_libjvm
- make func-test V=1
- make pytest-self-check
- make pytest-check;
```

Utility functions (which are not important build steps), such as `print_test_logs` and `setup_libjvm` can be placed into separate bash files. This way, an "executable build documentation" is what you get, and you only hide the non-important parts.

Reply:
Thanks for the review and a comment. My purpose with this change was the following:
if we can move the bash logic from .travis.yml into a well named bash script file we will be able to run it on our machine as well (if there are any problem during the Travis run and we need to debug something).